File: src/testingScripts/downloader.py
import json
import sys
import os
import urllib.request
import requests


def download_image(src, path , name):
    page = requests.get(src)
    with open(path,'wb') as f:
        f.write(page.content)
    print(name+' Downloaded')





filePath = os.path.join(os.getcwd(),'src','testingScripts','imgLinksFinal.json')
templateImages = os.path.join(os.getcwd(), 'public', 'images', 'templateImages')
previewImages = os.path.join( os.getcwd(),'public', 'images', 'previewImages')



images = json.load(open(filePath, ))


for k in images:
    title = images[k]['Title']
    imageLink = images[k]['Image Link']
    name = imageLink.split("/")[5]
    previewname = 'Preview-' + imageLink.split("/")[5]
    previewLink = images[k]['Preview Image Link']
    imageLocation = os.path.join(templateImages, name)
    previewLocation = os.path.join(previewImages, previewname)
    download_image(imageLink,imageLocation,name)
    download_image(previewLink,previewLocation,previewname)



print('Finished')

File: src/testingScripts/downloader.js
const fs = require('fs')
const path = require('path')
const request = require('request')
const https = require('https');
const download = require('image-downloader')

const filepath = path.join(__dirname, 'imgLinksFinal.json')
const templateImages = path.join(__dirname, '..', '..', 'public', 'images', 'templateImages')
const previewImages = path.join(__dirname, '..', '..', 'public', 'images', 'previewImages')


const images = JSON.parse(fs.readFileSync(filepath))


const downloadadaws = (url, path) => {
    request.head(url, (err, res, body) => {
        request(url)
            .pipe(fs.createWriteStream(path))
            .on('close', console.log('Done'))
    })
}

function saveImageToDisk(url, localPath) {
    var file = fs.createWriteStream(localPath);
    // var request = 
    https.get(url, function (response) {
        response.pipe(file);
    });
}





for (let k in images) {
    const title = images[k]['Title']
    const imageLink = images[k]['Image Link']
    const name = k + '-' + imageLink.split("/")[5]
    const previewname = k + '-preview-' + imageLink.split("/")[5]
    const previewLink = images[k]['Preview Image Link']
    const imageLocation = path.join(templateImages, name)
    const previewLocation = path.join(previewImages, previewname)
    const imageOption = {
        url: imageLink,
        dest: imageLocation
    }
    const previewOption = {
        url: previewLink,
        dest: previewLocation
    }
    download.image(imageOption)
        .then(({ filename }) => {
            console.log('Saved to', filename)  // saved to /path/to/dest/photo.jpg
        })
        .catch((err) => console.error(err))
    download.image(previewOption)
        .then(({ filename }) => {
            console.log('Saved to', filename)  // saved to /path/to/dest/photo.jpg
        })
        .catch((err) => console.error(err))
}



//Node.js Function to save image from External URL.
